### PR TITLE
[MIN-59] Update Streamlit application logic

### PR DESCRIPTION
**Pull request by dudeperf3ct**
In this PR, we update the  streamlit application logic to handle following features

1. Connecting to chromadb
2. Querying chromadb to search for similar documents to query
3. Prepare these context (question and similar documents) in a format expected by the LLM model (either QA or text generation format).
4. Display the answer response.

This trip has to happen twice for both mind and nhs data. This will enable streamlit application to communicate with chroma server.

### CODE REVIEW

File: app/app.py [hunk 1]
```diff
@@ -2,20 +2,34 @@
...
 
 import requests
 import streamlit as st
+from chromadb.api import API
+from chromadb.api.types import EmbeddingFunction
+from chromadb.utils import embedding_functions
+from utils.chroma_store import ChromaStore
+
+# Setup for chroma vector store
+CHROMA_SERVER_HOST_NAME = "localhost"  # "server.default"
```
Comment (dudeperf3ct) [inline app/app.py]: Change back to `server.default` once local testing is done.

File: README.md [hunk 2]
```diff
@@ -33,7 +33,7 @@ The repository for this project is one method where you can monitor progress - w
 
 ## Embedding pipeline
 
-To run the Zenml embedding pipeline, the resources such as AKS and ACR have to provisioned. AKS is used running Chroma server on AKS and ACR is used to store Chroma server image. 
+To run the Zenml embedding pipeline, the resources such as AKS and ACR have to provisioned. AKS is used running Chroma server on AKS and ACR is used to store Chroma server image.
```
Comment (JonoCX) [inline README.md]: To run the embedding pipeline, both AKS and ARC need to be provisioned. We use AKS to run the Chroma (vector database) service and ACT is used to host the Chroma server image.

File: README.md [hunk 3]
```diff
@@ -127,11 +138,11 @@ The expected payload structure is as follows:
 
 ## Streamlit Application
 
-Once the deployment has completed, run the following command to start a streamlit application. Note: you need to first obtain the model ingress IP as described above. 
+Once the deployment has completed, run the following command to start a streamlit application. Note: you need to first obtain the model ingress IP as described above.
```
Comment (JonoCX) [inline README.md]: Use the markdown tools available:

> Note: ...

File: app/app.py [hunk 4]
```diff
@@ -44,19 +58,85 @@ def _get_prediction_endpoint() -> Optional[str]:
...
+    """
+    try:
+        # Connect to vector store
+        chroma_client = ChromaStore(
+            chroma_server_hostname=chroma_server_host,
+            chroma_server_port=chroma_server_port,
+        )
+        return chroma_client
+    except Exception:
+        return None
```
Comment (JonoCX) [inline app/app.py:96]: Do we need to do something if an exception is thrown?

**Review by JonoCX — CHANGES_REQUESTED** [review]
Nice work! See comments

File: app/app.py (hunk 4 above)
Comment (dudeperf3ct) [reply app/app.py:96]: It's raised on line 244, the _streamlit way_

File: README.md [hunk 5]
```diff
@@ -33,7 +33,7 @@ The repository for this project is one method where you can monitor progress - w
 
 ## Embedding pipeline
 
-To run the Zenml embedding pipeline, the resources such as AKS and ACR have to provisioned. AKS is used running Chroma server on AKS and ACR is used to store Chroma server image. 
+To run the embedding pipeline, both AKS and ARC need to be provisioned. We use AKS to run the Chroma (vector database) service and ACT is used to host the Chroma server image.
```
Comment (d-lowl) [inline README.md]: I think both ARC and ACT mean ACR

**Review by d-lowl — APPROVED** [review]
The logic looks good to me for the initial pass. Nit in the readme

File: README.md (hunk 2 above)
Comment (d-lowl) [reply README.md]: I'm confused, what are ARC and ACT

**Review by JonoCX — COMMENTED** [review]
LGTM

**Review by JonoCX — APPROVED** [review]

**Pull request merged**